staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -910,6 +875,7 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 		klog.Warningf("Failed to derive event reference, won't create events: %v", err)
 		c.EventSink = nullEventSink{}
```

Comment:
nit: in the event that `eventReference()` returns  an error, `ref` will be nil, although i think we never see this in ci/production, otherwise the apiserver would be crash looping.

instead of changing the config `c.EventSink = nullEventSink{}`, we can change the server `s.eventSink = nullEventSink{}`

if we change the config, then  `s.RegisterDestroyFunc(c.EventSink.Destroy)` registers the `nullEventSink` instance, not on the  original instance.